packages/driver/src/cy/commands/waiting.js

```diff
@@ -162,11 +162,28 @@ module.exports = (Commands, Cypress, cy, state) => {
         log.set('referencesAlias', aliases)
       }
 
-      if (command && !['route', 'route2', 'intercept'].includes(command.get('name'))) {
-        $errUtils.throwErrByPath('wait.invalid_alias', {
-          onFail: options._log,
-          args: { alias },
-        })
+      const isNetworkInterceptCommand = (command) => {
+        const commandsThatCreateNetworkIntercepts = ['route', 'route2', 'intercept']
+        const commandName = command.get('name')
+
+        return commandsThatCreateNetworkIntercepts.includes(commandName)
+      }
+
+      const findNetworkAlias = (alias) => {
+        const routes = cy.state('routes') || {}
+
+        return _.find(_.values(routes), { alias })
+      }
```

Comment:
this doesn't cover dynamic aliases with `req.alias` i don't think

but luckily you can use `getAliasedRequests`: https://github.com/cypress-io/cypress/blob/c6f03e390f2b3c55f9c3b9a08611c6aabad2d607/packages/driver/src/cy/net-stubbing/aliasing.ts#L14-L28

Reply:
Seems to work as is, I have added a test to confirm it